src/services/navigationBar.ts

```diff
@@ -424,10 +424,10 @@ module ts.NavigationBar {
             }
 
             function createFunctionItem(node: FunctionDeclaration) {
-                if (node.name && node.body && node.body.kind === SyntaxKind.Block) {
+                if ((node.name || node.flags & NodeFlags.Default) && node.body && node.body.kind === SyntaxKind.Block) {
                     var childItems = getItemsWorker(sortNodes((<Block>node.body).statements), createChildItem);
 
-                    return getNavigationBarItem(node.name.text,
+                    return getNavigationBarItem((node.flags & NodeFlags.Default) ? "default": node.name.text ,

```

Comment:
Stray space; also I think this was a little cleaner when you just used an intermediate variable.